### DISCUSSION TITLE
pcode offset hightlight

### DISCUSSION THREAD

**Discussion by mksonzien** [HackOvert/GhidraSnippets#2] [Q&A]
Hello
I want to highlight only a specific address out of the whole function pcode.
Please inquire if it is possible to implement this.

def getAddress(offset):
    return currentProgram.getAddressFactory().getDefaultAddressSpace().getAddress(offset)

addr = getAddress(0x00112ff1)
func_name = functionManager.getFunctionContaining(addr)
function = getGlobalFunctions(str(func_name))[0]
results = ifc.decompileFunction(function, 0, ConsoleTaskMonitor())
pcode = results.getDecompiledFunction().getC()
// highlight(pcode, offset)

**Comment by cetfor** [comment]
Hey - sorry I'm so late to this.

I don't believe there is a way to highlight or colorize the decompiler output.  You can set secondary highlights on each node, but that ends up highlighting nodes all over the decompile window.

You should be able to highlight a line in the "Listing" (disassembly) view, similar to right-clicking, selecting "Colors" then "Set Color".  Let me know if that would be helpful.  I wish there were some way to highlight pseudo C lines, but I haven't found any way to accomplish that yet.
